salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
I setup a debian (closest to ubuntu I have) minion with a pip installed salt 2016.3.3 using your exact (minus device path) sls and I don't get the error.


```
root@kawaii:/etc/salt# salt-call --local -l debug -c /data/salt state.apply test
[DEBUG   ] Missing configuration file: /data/salt/minion
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: kawaii.UGent.be
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Please install 'virt-what' to improve results of the 'virtual' grain.
[WARNING ] /usr/local/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.

[DEBUG   ] Determining pillar cache
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded state.apply
[DEBUG   ] LazyLoaded saltutil.is_running
[DEBUG   ] LazyLoaded grains.get
[DEBUG   ] LazyLoaded roots.envs
[DEBUG   ] Could not LazyLoad roots.init
[DEBUG   ] Updating roots fileserver cache
[DEBUG   ] Determining pillar cache
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] Could not find file from saltenv 'base', 'salt://test.sls'
[DEBUG   ] Could not find file from saltenv 'base', 'salt://test/init.sls'
[DEBUG   ] compile template: False
[ERROR   ] Template was specified incorrectly: False
[DEBUG   ] LazyLoaded highstate.output
[ERROR   ] Data passed to highstate outputter is not a valid highstate return: {'local': ["No matching sls found for 'test' in env 'base'"]}
root@kawaii:/etc/salt# vim /etc/salt/minion
root@kawaii:/etc/salt# salt-call --local -l debug -c /etc/salt state.apply test
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Please install 'virt-what' to improve results of the 'virtual' grain.
[WARNING ] /usr/local/lib/python2.7/dist-packages/salt/grains/core.py:1493: DeprecationWarning: The "osmajorrelease" will be a type of an integer.

[DEBUG   ] Determining pillar cache
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded state.apply
[DEBUG   ] LazyLoaded saltutil.is_running
[DEBUG   ] LazyLoaded grains.get
[DEBUG   ] LazyLoaded roots.envs
[DEBUG   ] Could not LazyLoad roots.init
[DEBUG   ] Updating roots fileserver cache
[DEBUG   ] Determining pillar cache
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] In saltenv 'base', looking at rel_path 'test.sls' to resolve 'salt://test.sls'
[DEBUG   ] In saltenv 'base', ** considering ** path '/var/cache/minion/files/base/test.sls' to resolve 'salt://test.sls'
[DEBUG   ] Fetching file from saltenv 'base', ** attempting ** 'salt://test.sls'
[DEBUG   ] No dest file found 
[INFO    ] Fetching file from saltenv 'base', ** done ** 'test.sls'
[DEBUG   ] compile template: /var/cache/minion/files/base/test.sls
[DEBUG   ] Jinja search path: ['/var/cache/minion/files/base']
[PROFILE ] Time (in seconds) to render '/var/cache/minion/files/base/test.sls' using 'jinja' renderer: 0.00333499908447
[DEBUG   ] Rendered data from file: /var/cache/minion/files/base/test.sls:
# vim:foldmethod=indent:foldlevel=1:foldignore="":

zfsroot:
  zpool.present:
    - name: zfsroot
    - config:
        import: false
        force: true
    - layout:
        - /tmp/disk0

[DEBUG   ] LazyLoaded config.get
[DEBUG   ] Results of YAML rendering: 
OrderedDict([('zfsroot', OrderedDict([('zpool.present', [OrderedDict([('name', 'zfsroot')]), OrderedDict([('config', OrderedDict([('import', False), ('force', True)]))]), OrderedDict([('layout', ['/tmp/disk0'])])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/minion/files/base/test.sls' using 'yaml' renderer: 0.00642895698547
[DEBUG   ] LazyLoaded zpool.create
[DEBUG   ] LazyLoaded zpool.present
[INFO    ] Running state [zfsroot] at time 13:20:04.290867
[INFO    ] Executing state zpool.present for zfsroot
[DEBUG   ] zpool.present::zfsroot::config - {'device_dir': '/dev', 'force': True, 'import_dirs': None, 'import': False}
[DEBUG   ] zpool.present::zfsroot::layout - ['/tmp/disk0']
[DEBUG   ] LazyLoaded cmd.run_all
[INFO    ] Executing command '/sbin/zpool list zfsroot' in directory '/root'
[ERROR   ] Command '/sbin/zpool list zfsroot' failed with return code: 1
[ERROR   ] stderr: cannot open 'zfsroot': no such pool
[ERROR   ] retcode: 1
[DEBUG   ] zpool.present::zfsroot::creating
[INFO    ] Executing command '/sbin/zpool list zfsroot' in directory '/root'
[ERROR   ] Command '/sbin/zpool list zfsroot' failed with return code: 1
[ERROR   ] stderr: cannot open 'zfsroot': no such pool
[ERROR   ] retcode: 1
[INFO    ] Executing command '/sbin/zpool create -f zfsroot /tmp/disk0' in directory '/root'
[INFO    ] {'zfsroot': 'created'}
[INFO    ] Completed state [zfsroot] at time 13:20:04.534380 duration_in_ms=243.513
[DEBUG   ] File /var/cache/minion/accumulator/139875130949136 does not exist, no need to cleanup.
[DEBUG   ] LazyLoaded config.option
[DEBUG   ] LazyLoaded highstate.output
[DEBUG   ] LazyLoaded nested.output
local:
----------
          ID: zfsroot
    Function: zpool.present
      Result: True
     Comment: storage pool zfsroot was created
     Started: 13:20:04.290867
    Duration: 243.513 ms
     Changes:   
              ----------
              zfsroot:
                  created

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 243.513 ms
```

Can you give it a shot without your changes?